File: backend/routes/news.py
"""
News API routes
"""

from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import Optional
from services.news_parser import NewsParser
from services.llm import LLMService

router = APIRouter()


class AdaptNewsRequest(BaseModel):
    content: str
    level: str
    model: Optional[str] = None
    target_language: str = "German"


@router.get("/categories")
async def get_categories():
    """Get available news categories"""
    try:
        categories = NewsParser.get_categories()
        return {"categories": categories}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/{category}")
async def get_news(category: str, limit: int = 10):
    """Get news articles for a category"""
    try:
        articles = NewsParser.fetch_news(category, limit)
        return {"articles": articles}
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/adapt")
async def adapt_news(request: AdaptNewsRequest):
    """Adapt a news article to a target level"""
    try:
        result = await LLMService.adapt_content(
            text=request.content,
            level=request.level,
            model=request.model,
            target_language=request.target_language
        )
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

File: src/services/api/news.js
/**
 * News API Service
 * Handles news fetching and adaptation
 */

import { get, post } from './client';

/**
 * Get available news categories
 */
export async function getNewsCategories() {
    try {
        const response = await get('/news/categories');
        return response.categories;
    } catch (error) {
        console.error('Failed to fetch news categories:', error);
        throw error;
    }
}

/**
 * Get news articles for a category
 */
export async function getNewsByCategory(category, limit = 10) {
    try {
        const response = await get(`/news/${category}`, { limit });
        return response.articles;
    } catch (error) {
        console.error(`Failed to fetch news for category ${category}:`, error);
        throw error;
    }
}

/**
 * Adapt a news article to a target level
 */
export async function adaptNewsArticle({ content, level, model = null, targetLanguage = 'German' }) {
    try {
        const response = await post('/news/adapt', {
            content,
            level,
            model,
            target_language: targetLanguage,
        });
        return response;
    } catch (error) {
        console.error('News adaptation failed:', error);
        throw error;
    }
}

export default {
    getNewsCategories,
    getNewsByCategory,
    adaptNewsArticle,
};
